### PR TITLE
fix: paginator and null cursor values

### PR DESCRIPTION
**Pull request by dgvncsz0f**
This patch fixed the paginator when cursor values can be nil. We
assume that the order by will always follow postgres defaults, which
are:

  * asc: asc_nulls_last
  * desc: desc_nulls_first

Which means that other options are not supported, namely,
:asc_nulls_first and :desc_nulls_last.

Previously, the code would remove from the where clause all fields
which the cursor value is nil. While this works for pagination using a
single field, it breaks if you use two or more fields.

This patch looks at the value instead and modifies the where
expression accordingly - depending on the whether the value is null or
not.

### CODE REVIEW

File: lib/paginator/config.ex [hunk 1]
```diff
@@ -70,7 +70,7 @@ defmodule Paginator.Config do
 
   defp cursor_values_match_cursor_fields?(cursor_values, cursor_fields) do
     cursor_keys = cursor_values |> Map.keys() |> Enum.sort()
-    sorted_cursor_fields = cursor_fields |> Keyword.keys() |> Enum.sort()
+    sorted_cursor_fields = cursor_fields |> Enum.map(&elem(&1, 0)) |> Enum.sort()
```
Comment (dgvncsz0f) [inline lib/paginator/config.ex:73]: keyword.keys is throwing an exception because the first element is not always an atom.

**Comment by dgvncsz0f** [comment]
I've also updated all libs. The current versions were producing too many warnings when running the tests.

File: test/paginator_test.exs [hunk 2]
```diff
@@ -9,8 +9,14 @@ defmodule PaginatorTest do
...
 
   test "paginates forward", %{
-    payments: {p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11, p12}
+    payments: {p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, p11, p12} = payments
   } do
+    payments
+    |> Tuple.to_list()
+    |> Enum.with_index()
+    |> Enum.map(fn {p, i} -> {NaiveDateTime.to_iso8601(p.charged_at), p.id, "p#{i + 1}"} end)
+    |> Enum.sort()
```
Comment (c-brenn) [inline test/paginator_test.exs:18]: Is the result of this expression used?

**Review by c-brenn — APPROVED** [review]

File: test/paginator_test.exs (hunk 2 above)
Comment (dgvncsz0f) [reply test/paginator_test.exs:18]: 🙈 I was debugging something and forget to remove it. Doing that now.

**Pull request merged**